collect_app/src/main/java/org/odk/collect/android/tasks/InstanceServerUploader.java

```diff
@@ -160,17 +160,19 @@ private boolean uploadOneSubmission(String urlString, String id, String instance
             }
 
             // if https then enable preemptive basic auth...
-            if (u.getScheme() != null && u.getScheme().equals("https")) {
-                WebUtils.enablePreemptiveBasicAuth(localContext, u.getHost());
+            if (submissionUri.getScheme() != null && submissionUri.getScheme().equals("https")) {
+                WebUtils.enablePreemptiveBasicAuth(localContext, submissionUri.getHost());
             }
 
-            // we need to issue a head request
-            HttpHead httpHead = WebUtils.createOpenRosaHttpHead(u);
+            // Issue a head request to confirm the server is an OpenRosa server and see if auth
+            // is required
+            // http://docs.opendatakit.org/openrosa-form-submission/#extended-transmission-considerations
+            HttpHead httpHead = WebUtils.createOpenRosaHttpHead(submissionUri);
 
             // prepare response
-            HttpResponse response = null;
+            HttpResponse response;
```

Comment:
Making this a blank final will simplify understanding the code.

Reply:
As amazing as IDEA is, I’m surprised it doesn’t offer a change to a blank final for these redundant initializations.